### PR TITLE
Document more details on validation

### PR DESCRIPTION
**Pull request by GretaCB**
Per https://github.com/mapbox/vtvalidate/issues/23 and https://github.com/mapbox/vtvalidate/issues/25

- Add more documentation on what validation is happening and how that relates to vtzero and the VT spec.
- Remove API.md

cc @mapsam @springmeyer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,19 @@ make distclean
 ```
 
 ## Usage
+ 
+- If the tile is valid, `vtvalidate` will return an empty string. 
+- If the tile is invalid, `vtvalidate` will return the string output from vtzero. 
+- `vtvalidate` will throw an error if there is unexpected behaviour, for example an invalid parameter value or a corrupt compressed buffer.
```
Comment (mapsam) [inline README.md]: @GretaCB what is a "parameter value" in this case?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,19 @@ make distclean
...
+
+#### Type of validation
+`vtvalidate` validates tile data against [vtzero](https://github.com/mapbox/vtzero):
+- Tile data consistent with the [Mapbox vector tile spec - Version 2](https://www.mapbox.com/vector-tiles/specification/). Tiles created via Mapbox vector tile spec Version 1 will be flagged invalid.
```
Comment (mapsam) [inline README.md]: Love it - super explicit 👍

File: README.md [hunk 3]
```diff
@@ -25,6 +25,19 @@ make distclean
...
+- Read tile layer(s) and feature(s)
+- Decode properties
+- Decode geometries
```
Comment (mapsam) [inline README.md]: Does validate decode and confirm they are OGC Simple? Or just able to decode and not doing the extra leg lifting of OGC simple check?

File: README.md (hunk 1 above)
Comment (GretaCB) [reply README.md]: Fixed!

File: README.md (hunk 3 above)
Comment (GretaCB) [reply README.md]: Fixed!

**Pull request merged**